SMB Anonymous Login

Tracked by #15065 

Gathered Logs:
[15069-guest.txt](https://github.com/iterate-ch/cyberduck/files/12475441/15069-guest.txt)
[15069-anonymous.txt](https://github.com/iterate-ch/cyberduck/files/12475443/15069-anonymous.txt)

```
2023-08-30 13:53:13,833 [background-5] INFO  com.hierynomus.smbj.connection.PacketEncryptor - Initialized PacketEncryptor with Cipher << null >>
2023-08-30 13:53:13,834 [background-5] INFO  com.hierynomus.smbj.connection.Connection - Successfully connected to: diskstation.lan.xc
2023-08-30 13:53:13,834 [background-5] DEBUG ch.cyberduck.core.smb.SMBSession - Connected to ConnectionContext{
  serverGuid=6b736964-7473-7461-696f-6e0000000000,
  serverName='diskstation.lan.xc',
  negotiatedProtocol=NegotiatedProtocol{dialect=SMB_3_1_1, maxTransactSize=8388608, maxReadSize=8388608, maxWriteSize=8388608},
  clientGuid=dab1bee0-1e98-4d99-bd12-b8e2cd5949a1,
  clientCapabilities=[SMB2_GLOBAL_CAP_DFS, SMB2_GLOBAL_CAP_LARGE_MTU],
  serverCapabilities=[SMB2_GLOBAL_CAP_DFS, SMB2_GLOBAL_CAP_LEASING, SMB2_GLOBAL_CAP_LARGE_MTU],
  clientSecurityMode=1,
  serverSecurityMode=1,
  server='com.hierynomus.smbj.server.Server@2d86638'
}
2023-08-30 13:53:13,834 [background-5] DEBUG ch.cyberduck.core.Session - Connection did open to Host{protocol=Profile{parent=smb, vendor=iterate GmbH, description=SMB (Server Message Block), image=null}, region='null', port=445, hostname='diskstation.lan.xc', credentials=Credentials{user='anonymous', password='', oauth='OAuthTokens{accessToken='', refreshToken='', expiryInMilliseconds=9223372036854775807}', token='', identity=null}, uuid='f57fabd5-ba6c-43c1-ade8-d791b6b3bcc0', nickname='Diskstation SMB', defaultpath='null', workdir=null, custom=null, labels=null}
2023-08-30 13:53:13,836 [background-5] DEBUG ch.cyberduck.core.KeychainLoginService - Attempt authentication for Session{host=Host{protocol=Profile{parent=smb, vendor=iterate GmbH, description=SMB (Server Message Block), image=null}, region='null', port=445, hostname='diskstation.lan.xc', credentials=Credentials{user='anonymous', password='', oauth='OAuthTokens{accessToken='', refreshToken='', expiryInMilliseconds=9223372036854775807}', token='', identity=null}, uuid='f57fabd5-ba6c-43c1-ade8-d791b6b3bcc0', nickname='Diskstation SMB', defaultpath='null', workdir=null, custom=null, labels=null}, state=open}
2023-08-30 13:53:13,847 [background-5] DEBUG com.hierynomus.smbj.auth.NtlmAuthenticator - Initialized Authentication of Guest using NTLM
2023-08-30 13:53:13,861 [background-5] DEBUG com.hierynomus.smbj.connection.Connection - Granted 1 (out of 1) credits to SMB2_SESSION_SETUP with message id << 1 >>
2023-08-30 13:53:13,861 [background-5] DEBUG com.hierynomus.smbj.transport.tcp.direct.DirectTcpTransport - Writing packet SMB2_SESSION_SETUP with message id << 1 >>
2023-08-30 13:53:13,862 [background-5] DEBUG com.hierynomus.protocol.commons.concurrent.Promise - Awaiting << 1 >>
2023-08-30 13:53:13,864 [Packet Reader for diskstation.lan.xc] DEBUG com.hierynomus.smbj.transport.tcp.direct.DirectTcpPacketReader - Received packet SMB2_SESSION_SETUP with message id << 1 >>
2023-08-30 13:53:13,864 [Packet Reader for diskstation.lan.xc] DEBUG com.hierynomus.smbj.connection.packet.SMB2CreditGrantingPacketHandler - Server granted us 1 credits for SMB2_SESSION_SETUP with message id << 1 >>, now available: 1 credits
2023-08-30 13:53:13,865 [Packet Reader for diskstation.lan.xc] DEBUG com.hierynomus.protocol.commons.concurrent.Promise - Setting << 1 >> to `SMB2_SESSION_SETUP with message id << 1 >>`
2023-08-30 13:53:13,867 [background-5] DEBUG com.hierynomus.smbj.connection.SMBSessionBuilder - More processing required for authentication of Guest using com.hierynomus.smbj.auth.NtlmSealer@2b48676
2023-08-30 13:53:13,868 [background-5] DEBUG com.hierynomus.smbj.auth.NtlmAuthenticator - Received token: a1 81 cc 30 81 c9 a0 03 0a 01 01 a1 0c 06 0a 2b 06 01 04 01 82 37 02 02 0a a2 81 b3 04 81 b0 4e 54 4c 4d 53 53 50 00 02 00 00 00 16 00 16 00 38 00 00 00 15 82 8a 62 a7 3f 7f 20 f1 81 d6 de 00 00 00 00 00 00 00 00 62 00 62 00 4e 00 00 00 06 01 00 00 00 00 00 0f 44 00 49 00 53 00 4b 00 53 00 54 00 41 00 54 00 49 00 4f 00 4e 00 02 00 16 00 44 00 49 00 53 00 4b 00 53 00 54 00 41 00 54 00 49 00 4f 00 4e 00 01 00 16 00 44 00 49 00 53 00 4b 00 53 00 54 00 41 00 54 00 49 00 4f 00 4e 00 04 00 00 00 03 00 16 00 64 00 69 00 73 00 6b 00 73 00 74 00 61 00 74 00 69 00 6f 00 6e 00 07 00 08 00 ee 55 93 8e 38 db d9 01 00 00 00 00
2023-08-30 13:53:13,872 [background-5] DEBUG com.hierynomus.asn1.ASN1InputStream - Read ASN.1 object: ASN1TaggedObject[ASN1Tag[CONTEXT_SPECIFIC,CONSTRUCTED,1],<unknown>]
2023-08-30 13:53:13,875 [background-5] DEBUG com.hierynomus.asn1.ASN1InputStream - Read ASN.1 object: ASN1TaggedObject[ASN1Tag[CONTEXT_SPECIFIC,CONSTRUCTED,0],<unknown>]
2023-08-30 13:53:13,875 [background-5] DEBUG com.hierynomus.asn1.ASN1InputStream - Read ASN.1 object: ASN1TaggedObject[ASN1Tag[CONTEXT_SPECIFIC,CONSTRUCTED,1],<unknown>]
2023-08-30 13:53:13,875 [background-5] DEBUG com.hierynomus.asn1.ASN1InputStream - Read ASN.1 object: ASN1TaggedObject[ASN1Tag[CONTEXT_SPECIFIC,CONSTRUCTED,2],<unknown>]
2023-08-30 13:53:13,875 [background-5] DEBUG com.hierynomus.asn1.ASN1InputStream - Read ASN.1 object: ASN1Sequence[[ASN1TaggedObject[ASN1Tag[CONTEXT_SPECIFIC,CONSTRUCTED,0],<unknown>], ASN1TaggedObject[ASN1Tag[CONTEXT_SPECIFIC,CONSTRUCTED,1],<unknown>], ASN1TaggedObject[ASN1Tag[CONTEXT_SPECIFIC,CONSTRUCTED,2],<unknown>]]]
2023-08-30 13:53:13,880 [background-5] DEBUG com.hierynomus.asn1.ASN1InputStream - Read ASN.1 object: ASN1Enumerated[1]
2023-08-30 13:53:13,882 [background-5] DEBUG com.hierynomus.asn1.ASN1InputStream - Read ASN.1 object: ASN1ObjectIdentifier[1.3.6.1.4.1.311.2.2.10]
2023-08-30 13:53:13,883 [background-5] DEBUG com.hierynomus.asn1.ASN1InputStream - Read ASN.1 object: ASN1OctetString[[78, 84, 76, 77, 83, 83, 80, 0, 2, 0, 0, 0, 22, 0, 22, 0, 56, 0, 0, 0, 21, -126, -118, 98, -89, 63, 127, 32, -15, -127, -42, -34, 0, 0, 0, 0, 0, 0, 0, 0, 98, 0, 98, 0, 78, 0, 0, 0, 6, 1, 0, 0, 0, 0, 0, 15, 68, 0, 73, 0, 83, 0, 75, 0, 83, 0, 84, 0, 65, 0, 84, 0, 73, 0, 79, 0, 78, 0, 2, 0, 22, 0, 68, 0, 73, 0, 83, 0, 75, 0, 83, 0, 84, 0, 65, 0, 84, 0, 73, 0, 79, 0, 78, 0, 1, 0, 22, 0, 68, 0, 73, 0, 83, 0, 75, 0, 83, 0, 84, 0, 65, 0, 84, 0, 73, 0, 79, 0, 78, 0, 4, 0, 0, 0, 3, 0, 22, 0, 100, 0, 105, 0, 115, 0, 107, 0, 115, 0, 116, 0, 97, 0, 116, 0, 105, 0, 111, 0, 110, 0, 7, 0, 8, 0, -18, 85, -109, -114, 56, -37, -39, 1, 0, 0, 0, 0]]
2023-08-30 13:53:13,886 [background-5] DEBUG com.hierynomus.ntlm.messages.NtlmChallenge - Windows version = WindowsVersion[WINDOWS_MAJOR_VERSION_6, WINDOWS_MINOR_VERSION_1, 0, NTLMSSP_REVISION_W2K3]
2023-08-30 13:53:13,891 [background-5] DEBUG com.hierynomus.smbj.auth.NtlmAuthenticator - Received NTLM challenge from: DISKSTATION
2023-08-30 13:53:13,912 [background-5] DEBUG com.hierynomus.smbj.auth.NtlmSealer - Calculating signing and sealing keys for NTLM Extended Session Security
2023-08-30 13:53:13,913 [background-5] DEBUG com.hierynomus.smbj.auth.NtlmSealer - Signing with NTLM Extended Session Security
2023-08-30 13:53:13,914 [background-5] DEBUG com.hierynomus.smbj.connection.Connection - Granted 1 (out of 1) credits to SMB2_SESSION_SETUP with message id << 2 >>
2023-08-30 13:53:13,914 [background-5] DEBUG com.hierynomus.smbj.transport.tcp.direct.DirectTcpTransport - Writing packet SMB2_SESSION_SETUP with message id << 2 >>
2023-08-30 13:53:13,914 [background-5] DEBUG com.hierynomus.protocol.commons.concurrent.Promise - Awaiting << 2 >>
2023-08-30 13:53:13,943 [Packet Reader for diskstation.lan.xc] DEBUG com.hierynomus.smbj.transport.tcp.direct.DirectTcpPacketReader - Received packet SMB2_SESSION_SETUP with message id << 2 >>
2023-08-30 13:53:13,943 [Packet Reader for diskstation.lan.xc] DEBUG com.hierynomus.smbj.connection.packet.SMB2CreditGrantingPacketHandler - Server granted us 511 credits for SMB2_SESSION_SETUP with message id << 2 >>, now available: 511 credits
2023-08-30 13:53:13,943 [Packet Reader for diskstation.lan.xc] DEBUG com.hierynomus.protocol.commons.concurrent.Promise - Setting << 2 >> to `SMB2_SESSION_SETUP with message id << 2 >>`
2023-08-30 13:53:13,949 [background-5] WARN  ch.cyberduck.core.threading.BackgroundCallable - Failure running background task WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=0}}}
java.lang.Exception: null
	at ch.cyberduck.core.threading.BackgroundCallable.<init>(BackgroundCallable.java:38) ~[Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at ch.cyberduck.core.threading.DefaultBackgroundExecutor.execute(DefaultBackgroundExecutor.java:81) ~[Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at ch.cyberduck.core.AbstractController.background(AbstractController.java:72) ~[Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at cli.Ch.Cyberduck.Ui.Controller.BrowserController$$$003C$$003Ec__DisplayClass229_0.<UnmountImpl>b__0(Unknown Source) ~[Cyberduck.exe:?]
	at ch.cyberduck.core.threading.BackgroundCallable$1.run(BackgroundCallable.java:76) ~[Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at cli.System.Delegate.DynamicInvokeImpl(Unknown Source) ~[mscorlib.dll:?]
	at cli.System.Windows.Forms.Control.InvokeMarshaledCallbackDo(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.Control.InvokeMarshaledCallbackHelper(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Threading.ExecutionContext.RunInternal(Unknown Source) ~[mscorlib.dll:?]
	at cli.System.Threading.ExecutionContext.Run(Unknown Source) ~[mscorlib.dll:?]
	at cli.System.Threading.ExecutionContext.Run(Unknown Source) ~[mscorlib.dll:?]
	at cli.System.Windows.Forms.Control.InvokeMarshaledCallback(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.Control.InvokeMarshaledCallbacks(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.Control.WndProc(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.Form.WndProc(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.NativeWindow.Callback(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.Application$ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.Application$ThreadContext.RunMessageLoopInner(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.Application$ThreadContext.RunMessageLoop(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.Ch.Cyberduck.Ui.Program.Main(Unknown Source) ~[Cyberduck.exe:?]
Caused by: java.lang.NullPointerException
	at com.hierynomus.smbj.connection.SMBSessionBuilder.deriveKey(SMBSessionBuilder.java:324) ~[Cyberduck.Protocols.DLL:0.12.2]
	at com.hierynomus.smbj.connection.SMBSessionBuilder.deriveKeys(SMBSessionBuilder.java:290) ~[Cyberduck.Protocols.DLL:0.12.2]
	at com.hierynomus.smbj.connection.SMBSessionBuilder.setupSession(SMBSessionBuilder.java:174) ~[Cyberduck.Protocols.DLL:0.12.2]
	at com.hierynomus.smbj.connection.SMBSessionBuilder.setupSession(SMBSessionBuilder.java:152) ~[Cyberduck.Protocols.DLL:0.12.2]
	at com.hierynomus.smbj.connection.SMBSessionBuilder.establish(SMBSessionBuilder.java:119) ~[Cyberduck.Protocols.DLL:0.12.2]
	at com.hierynomus.smbj.connection.Connection.authenticate(Connection.java:197) ~[Cyberduck.Protocols.DLL:0.12.2]
	at ch.cyberduck.core.smb.SMBSession.login(SMBSession.java:122) ~[Cyberduck.Protocols.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at ch.cyberduck.core.KeychainLoginService.authenticate(KeychainLoginService.java:196) ~[Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at ch.cyberduck.core.LoginConnectionService.authenticate(LoginConnectionService.java:166) ~[Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at ch.cyberduck.core.LoginConnectionService.connect(LoginConnectionService.java:157) ~[Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at ch.cyberduck.core.LoginConnectionService.check(LoginConnectionService.java:100) ~[Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at ch.cyberduck.core.pool.StatelessSessionPool.borrow(StatelessSessionPool.java:56) ~[Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at ch.cyberduck.core.threading.SessionBackgroundAction.run(SessionBackgroundAction.java:115) ~[Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at ch.cyberduck.core.threading.SessionBackgroundAction$1.call(SessionBackgroundAction.java:106) ~[Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at ch.cyberduck.core.threading.DefaultRetryCallable.call(DefaultRetryCallable.java:52) ~[Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at ch.cyberduck.core.threading.SessionBackgroundAction.call(SessionBackgroundAction.java:108) ~[Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at ch.cyberduck.core.threading.BackgroundCallable.run(BackgroundCallable.java:96) [Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at ch.cyberduck.core.threading.BackgroundCallable.call(BackgroundCallable.java:60) [Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:625) [?:1.8.0]
	at ch.cyberduck.core.threading.NamedThreadFactory$1.run(NamedThreadFactory.java:59) [Cyberduck.Core.DLL:5ee8202624aa0639208f4f64991fb6007e99f37a]
	at java.lang.Thread.run(Thread.java:955) [?:1.8.0]
```

--- a/profiles/default/SMB.cyberduckprofile
+++ b/profiles/default/SMB.cyberduckprofile
@@ -34,6 +34,6 @@
         <key>Password Configurable</key>
         <true/>
         <key>Anonymous Configurable</key>
-        <false/>
+        <true/>
     </dict>
 </plist>

